src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp

```diff
@@ -620,21 +619,54 @@ void FwAutotuneAttitudeControl::saveGainsToParams()
 
 const Vector3f FwAutotuneAttitudeControl::getIdentificationSignal()
 {
-	if (_steps_counter > _max_steps) {
-		_signal_sign = (_signal_sign == 1) ? 0 : 1;
-		_steps_counter = 0;
 
-		if (_max_steps > 1) {
-			_max_steps--;
 
-		} else {
-			_max_steps = 5;
+	const hrt_abstime now = hrt_absolute_time();
+	const float dt = math::constrain((now - last_time_signal_generator_called) * 1e-6f,
+					 MIN_AUTO_TIMESTEP, MAX_AUTO_TIMESTEP);
+	last_time_signal_generator_called = now;
+
+	switch (_param_fw_sysid_signal_type.get()) {
+	case  static_cast<int32_t>(SignalType::kStep): {
+			if (_steps_counter > _max_steps) {
+				_signal_sign = (_signal_sign == 1) ? 0 : 1;
+				_steps_counter = 0;
+
+				if (_max_steps > 1) {
+					_max_steps--;
+
+				} else {
+					_max_steps = 5;
+				}
+			}
+
+			signal = float(_signal_sign) * _param_fw_at_sysid_amp.get();
 		}
+		break;
+
+	case static_cast<int32_t>(SignalType::kLinearSineSweep): {
+
+			signal = signal_generator::getLinearSineSweep(_param_fw_sysid_start_frequency.get(),
+					_param_fw_sysid_end_frequency.get(),
+					_param_fw_sysid_duration.get(), dt);
```

Comment:
shouldn't be `dt` but the time since the signal started

Reply:
Sorry